src/charm.py

```diff
@@ -60,6 +64,27 @@ def __init__(self, *args):
             self.unit.get_container(CONTAINER), User(name=PEBBLE_USER[0], group=PEBBLE_USER[1])
         )
 
+        # oauth
+        self.auth_proxy_relation_name = "auth-proxy"
+
+        self.auth_proxy = AuthProxyRequirer(
+            self, self.auth_proxy_config, self.auth_proxy_relation_name
+        )
+
+    @property
+    def auth_proxy_config(self) -> AuthProxyConfig:
+        """Configure the auth proxy relation."""
+        return AuthProxyConfig(
+            protected_urls=[
+                self.ingress.url if self.ingress.url is not None else "https://some-test-url.com"
```

Comment:
**todo** Uhm, I'm not sure I like to have some dummy strings with tests names in the charm code. Could we do something else here? if having an ingress is required, we could place the charm into block status and tell the user to relate with traefik first, or something like this, rather than getting a fake URL

Reply:
+1, I would only send the actual url when ingress gets ready/changed.
About blocked status: if spark is related with oathkeeper but is missing an ingress relation, oathkeeper will get into `BlockedStatus` (it checks with traefik whether a charm is eligible for proxy protection). It is a way to inform about the missing integration, but it's up to you if you want to block your charm as well.

Reply:
Done